File: Data Structures & Algorithms/same-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        stack1=[]
        stack2=[]
        def dfs(node, stack):
            if not node:
                stack.append(None)
                return
            stack.append(node.val)
            dfs(node.left, stack)
            dfs(node.right, stack)
        dfs(p, stack1)
        dfs(q, stack2)
        print(stack1, stack2)
        return stack1 == stack2

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    
    levelOrder(root) {
        let result = []
        let curr = []
        function bfs(root){
            if (!root) return [];
            
            const queue = [root]; 

            while (queue.length) {
                let lvlLength = queue.length
                for(let i = 0; i<lvlLength; i++){
                    const node = queue.shift()
                    curr.push(node.val)
                    if (node.left)  queue.push(node.left)
                    if (node.right) queue.push(node.right)
                }
                result.push([...curr])
                curr = []
            }
        }
        bfs(root)
        return result
    }
}
